nuxeo-ai-similar-content/src/test/java/org/nuxeo/ai/similar/content/operations/CountsOperationTest.java

```diff
@@ -0,0 +1,102 @@
+ *  limitations under the License.
+ *
+ *
+ * Contributors:
+ *    Andrei Nechaev
+ *
+ */
+
+package org.nuxeo.ai.similar.content.operations;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.nuxeo.ai.pipes.functions.PropertyUtils.FILE_CONTENT;
+import static org.nuxeo.ai.pipes.services.JacksonUtil.MAPPER;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ai.similar.content.operation.FindSimilar;
+import org.nuxeo.ai.similar.content.operation.GetAssetCounts;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.elasticsearch.test.RepositoryElasticSearchFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.extension.responsetemplating.ResponseTemplateTransformer;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+
+@RunWith(FeaturesRunner.class)
+@Features({ AutomationFeature.class, RepositoryElasticSearchFeature.class })
+@Deploy("org.nuxeo.ai.similar-content")
+@Deploy("org.nuxeo.ai.ai-model")
+@Deploy("org.nuxeo.ai.nuxeo-jwt-authenticator-core")
+@Deploy("org.nuxeo.ai.similar-content-test:OSGI-INF/cloud-client-test.xml")
+@Deploy("org.nuxeo.ai.similar-content-test:OSGI-INF/dedup-config-test.xml")
+@Deploy("org.nuxeo.ai.similar-content-test:OSGI-INF/disable-dedup-listener.xml")
+@Deploy("org.nuxeo.ecm.automation.server")
+public class CountsOperationTest {
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(
+            options().extensions(new ResponseTemplateTransformer(true)).port(5089));
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected AutomationService automationService;
+
+    @Test
+    public void shouldGetsAllCounts() throws OperationException, IOException {
+        DocumentModel fileDoc = session.createDocumentModel("/", "TestFile", "File");
+        fileDoc.setPropertyValue(FILE_CONTENT, (Serializable) Blobs.createBlob("this is a blob"));
+        fileDoc = session.createDocument(fileDoc);
+        session.save();
```

Comment:
here, please create multiple objects

Reply:
some of them may have dedup facet so we can ensure that numbers are correct for all types of counts